homeassistant/components/seventeentrack/sensor.py

```diff
@@ -297,6 +300,18 @@ def __init__(
                     for package in to_add
                 ], True)
 
+            # Remove archived packages from the entity registry:
+            to_remove = set(self.packages) - set(packages)
+            reg = await self._hass.helpers.entity_registry.async_get_registry()
+            for package in to_remove:
+                entity_id = reg.async_get_entity_id(
+                    self, 'sensor', 'seventeentrack',
+                    ENTITY_ID_TEMPLATE.format(
+                        self.account_id, package.tracking_number))
+                if not entity_id:
+                    continue
+                self._hass.async_create_task(reg.async_remove(entity_id))
```

Comment:
Is it only the entries in the entity registry we need to remove, not the entities themselves?

Reply:
Good point: as with delivered packages, archived ones should have their entities removed, as well. Will address.

Reply:
FYI, b0b37c5 should address this when the entity itself updates, rather than when the data object updates.